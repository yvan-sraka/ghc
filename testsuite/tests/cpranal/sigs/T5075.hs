-- | This module currently asserts that we trim CPR for local bindings
-- returning a sum. We can hopefully give @loop@ a CPR signature some day, but
-- we first have to fix #5075/#16570.
module T5075 where

-- Omission of the type signature is deliberate, otherwise we won't get a join
-- point (this is up to the desugarer, not sure why).
-- loop :: (Ord a, Num a) => a -> Either a b
loop x = case x < 10 of
  True -> Left x
  False -> loop (x*2)
