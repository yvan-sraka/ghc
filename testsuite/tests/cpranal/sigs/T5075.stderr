
==================== Cpr signatures ====================
T5075.$trModule: *
T5075.loop: *


